biocrnpyler/sbmlutil.py

```diff
@@ -230,9 +230,9 @@ def _create_products(product_list, sbml_reaction, model):
         product.setConstant(False)
 
 def _create_modifiers(crn_reaction, sbml_reaction, model):
-    reactants_list = [i.species.name for i in crn_reaction.inputs]
-    products_list = [i.species.name for i in crn_reaction.outputs]
-    modifier_species = [i.name for i in crn_reaction.propensity_type.propensity_dict['species'].values()]
+    reactants_list = [str(i.species) for i in crn_reaction.inputs]
+    products_list = [str(i.species) for i in crn_reaction.outputs]
+    modifier_species = [str(i) for i in crn_reaction.propensity_type.propensity_dict['species'].values()]
```

Comment:
So, these 3 line changes fixed the problem of species modifier reference in SBML to be set to correct species IDs rather than names, right?

just want to confirm here for the sake of documenting this.

Reply:
Yep - the new SBML files have the modifiers reference say "protein_A" instead of "A".